source/common/common/logger.cc

```diff
@@ -48,6 +52,21 @@ void StderrSinkDelegate::flush() {
   std::cerr << std::flush;
 }
 
+StderrEscapeNewlineSinkDelegate::StderrEscapeNewlineSinkDelegate(DelegatingLogSinkPtr log_sink)
+    : StderrSinkDelegate(log_sink) {}
+
+void StderrEscapeNewlineSinkDelegate::log(absl::string_view msg) {
+  Thread::OptionalLockGuard guard(lock_);
+  for (size_t i = 0; i < msg.size() - 1; i++) {
+    if (msg[i] == '\n') {
+      std::cerr << "\\n ";
+    } else {
+      std::cerr << msg[i];
+    }
+  }
+  std::cerr << msg[msg.size() - 1];
```

Comment:
I think this should be a utility and also used for the file log. Could we just make use of https://github.com/abseil/abseil-cpp/blob/master/absl/strings/escaping.h? This seems roughly what we're after.

Reply:
Good idea. This will do more than just escape new lines, but I'll clarify in the comments.